be/src/exec/vectorized/analytor.h

```diff
@@ -176,9 +175,14 @@ class Analytor final : public pipeline::ContextWithDependency {
     int64_t _found_partition_end = 0;
 
     // A peer group is all of the rows that are peers within the specified ordering.
-    // Rows are peers if they compare equal to each other using the specified ordering expression.
+    // Record the start pos of current peer group.
     int64_t _peer_group_start = 0;
+    // Record the end pos of current peer group.
+    // If the end position has not been found during the iteration, _peer_group_end = _peer_group_start.
     int64_t _peer_group_end = 0;
+    // Used to record the first position of the latest Chunk that is not equal to the peer group.
+    // If not found, it points to the last position + 1.
+    int64_t _found_peer_group_end = 0;
```

Comment:
Should we deal with the `_found_peer_group_end` in remove_unused_buffer_values ?

Reply:
Yes, it is. And currently `_found_peer_group_end` is unused outside the function `find_and_check_peer_group_end`, so my test do not find the problem. I'll make it up.